buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchNode.java

```diff
@@ -341,6 +345,49 @@ public void module(String moduleProjectPath) {
         module(maybeCreatePluginOrModuleDependency(moduleProjectPath));
     }
 
+    @Override
+    public void extension(ExtensionsProperties extensions) {
+        this.extensions.add(extensions);
+    }
+
+    public void writeExtensionFiles() {
+        try {
+            // Creates extensions.yml in the target directory
+            Path destination = getDistroDir().resolve("extensions").resolve("extensions.yml");
+            if (!Files.exists(getDistroDir().resolve("extensions"))) {
+                Files.createDirectory(getDistroDir().resolve("extensions"));
+            }
+            DumperOptions dumperOptions = new DumperOptions();
+            TestExtensionsList extensionsList = new TestExtensionsList(this.extensions);
+            dumperOptions.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+            Yaml yaml = new Yaml(dumperOptions);
+            Files.write(destination, yaml.dump(extensionsList).getBytes());
+
+            /*
+             * SnakeYaml creates a Yaml file with an unnecessary line at the top with the class name
+             * This section of code removes that line while keeping everything else the same.
+             */
```

Comment:
Can you try using `yaml.dumpAs` instead of `yaml.dump` to avoid this? Let's see if that works

Reply:
helloWorld is currently broken on the SDK, so I can't test out dumpAs at the moment.  I can definitely explore this in the future.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/7404